src/phylum/init/cli.py

```diff
@@ -0,0 +1,270 @@
+# should be produced. A target triple consists of three strings separated by a hyphen, with a possible fourth string
+# at the end preceded by a hyphen. The first is the architecture, the second is the "vendor", the third is the OS
+# type, and the optional fourth is environment type.
+#
+# References:
+#   * https://doc.rust-lang.org/nightly/rustc/platform-support.html
+#   * https://rust-lang.github.io/rfcs/0131-target-specification.html
+SUPPORTED_TARGET_TRIPLES = (
+    "aarch64-apple-darwin",
+    "x86_64-apple-darwin",
+    "x86_64-unknown-linux-musl",
+)
+# Keys are lowercase machine hardware names as returned from `uname -m`.
+# Values are the mapped rustc architecture.
+SUPPORTED_ARCHES = {
+    "arm64": "aarch64",
+    "amd64": "x86_64",
+}
+# Keys are lowercase operating system name as returned from `uname -s`.
+# Values are the mapped rustc platform, which is the vendor-os_type[-environment_type].
+SUPPORTED_PLATFORMS = {
+    "linux": "unknown-linux-musl",
+    "darwin": "apple-darwin",
+}
+
+TOKEN_ENVVAR_NAME = "PHYLUM_TOKEN"
+PHYLUM_PATH = pathlib.Path.home() / ".phylum"
+PHYLUM_BIN_PATH = PHYLUM_PATH / "phylum"
+SETTINGS_YAML_PATH = PHYLUM_PATH / "settings.yaml"
+
+# Potential features to add:
+#  * Add logging support, a verbosity option to control it, and swap out print statements for logging
+#  * Check for valid versions by using the GitHub API to compare against actual releases
+#    * If so, also programmatically generate the SUPPORTED_TARGET_TRIPLES
+#  * Use the GitHub API ("https://api.github.com") to more programmatically get the archive URL
+#  * Add a `--list` option, to show which versions are available
+#  * Add an option to account for pre-releases
+
+
+def version_check(version):
+    """Check a given version for validity and return a normalized form of it."""
+    if version == "latest":
+        return version
+
+    version = version.lower()
+    if not version.startswith("v"):
+        version = f"v{version}"
+
+    try:
+        # The release layout structure changed starting with v2.0.0 and support here is only for the new layout
+        if Version("v2.0.0") > Version(canonicalize_version(version)):
+            raise argparse.ArgumentTypeError("version must be at least v2.0.0")
+    except InvalidVersion as err:
+        raise argparse.ArgumentTypeError("an invalid version was provided") from err
+
+    return version
+
+
+def get_target_triple():
+    """Get the "target triple" from the current system and return it."""
+    arch = SUPPORTED_ARCHES.get(platform.uname().machine.lower(), "unknown")
+    plat = SUPPORTED_PLATFORMS.get(platform.uname().system.lower(), "unknown")
+    return f"{arch}-{plat}"
+
+
+def save_file_from_url(url, path):
+    """Save a file from a given URL to a local file path, in binary mode."""
+    print(f" [*] Getting {url} file ...", end="")
+    req = requests.get(url, timeout=2.0)
```

Comment:
A 2 second timeout seems pretty strict... Do we really need it to be that fast?

Reply:
It can be longer, for sure. Is there a suggestion? Maybe 5 seconds. Keep in mind this note about timeouts, from the [`requests` quickstart guide](https://docs.python-requests.org/en/latest/user/quickstart/#timeouts):

> timeout is not a time limit on the entire response download; rather, an exception is raised if the server has not issued a response for timeout seconds (more precisely, if no bytes have been received on the underlying socket for timeout seconds). If no timeout is specified explicitly, requests do not time out.

Reply:
5 is good

Reply:
It has been changed to 5.0 seconds.